pkg/fileutils/fileutils_linux.go

```diff
@@ -0,0 +1,18 @@
+package fileutils
+
+import (
+	"os"
+
+	"github.com/docker/docker/pkg/system"
+)
+
+// CurrentUserIsOwner checks whether the current user is the owner of the given
+// file.
+func CurrentUserIsOwner(f string) (bool, error) {
```

Comment:
only code path that return `false, err` is here:
https://github.com/docker/docker/pull/11909/files#diff-16d4f31957002f2aca808fe16c6d9c7dR9

I agree with you that's not so good returning err always nil, I did that only because I want to be explicit about raising an error if someone is using that function with windows.
Returning false sounds like ok that user is not the owner of the file not I cannot stat the file.